analyzer_test.go

```diff
@@ -1,13 +1,649 @@
+				Platform: platform.NewPlatform(apiString),
+				Buildpacks: []buildpack.GroupBuildpack{
+					{ID: "metadata.buildpack", API: api.Buildpack.Latest().String()},
+					{ID: "no.cache.buildpack", API: api.Buildpack.Latest().String()},
+					{ID: "no.metadata.buildpack", API: api.Buildpack.Latest().String()},
+				},
+				Cache:                 testCache,
+				LayerMetadataRestorer: metadataRestorer,
+			}
+			if testing.Verbose() {
+				analyzer.Logger = cmd.DefaultLogger
+				h.AssertNil(t, cmd.SetLogLevel("debug"))
+			}
+			mockCtrl = gomock.NewController(t)
+		})
+
+		it.After(func() {
+			h.AssertNil(t, os.RemoveAll(tmpDir))
+			h.AssertNil(t, os.RemoveAll(layersDir))
+			h.AssertNil(t, os.RemoveAll(cacheDir))
+			h.AssertNil(t, image.Cleanup())
+			mockCtrl.Finish()
+		})
+
+		when("#Analyze", func() {
+			var (
+				expectedAppMetadata platform.LayersMetadata
+				ref                 *testmock.MockReference
+			)
+
+			it.Before(func() {
+				ref = testmock.NewMockReference(mockCtrl)
+				ref.EXPECT().Name().AnyTimes()
+			})
+
+			when("image exists", func() {
+				it.Before(func() {
+					metadata := h.MustReadFile(t, filepath.Join("testdata", "analyzer", "app_metadata.json"))
+					h.AssertNil(t, image.SetLabel("io.buildpacks.lifecycle.metadata", string(metadata)))
+					h.AssertNil(t, json.Unmarshal(metadata, &expectedAppMetadata))
+				})
+
+				it("restores layer metadata and unsets the launch, build and cache flags", func() {
+					h.SkipIf(t, api.MustParse(apiString).Compare(api.MustParse("0.7")) >= 0, "Platform API >= 0.7 does not restore layer metadata")
+
+					_, err := analyzer.Analyze()
+					h.AssertNil(t, err)
+
+					unsetFlags := "[types]"
+					for _, data := range []struct{ name, want string }{
+						{"metadata.buildpack/launch.toml", "[metadata]\n  launch-key = \"launch-value\""},
+						{"metadata.buildpack/launch-build-cache.toml", "[metadata]\n  launch-build-cache-key = \"launch-build-cache-value\""},
+						{"metadata.buildpack/launch-cache.toml", "[metadata]\n  launch-cache-key = \"launch-cache-value\""},
+						{"no.cache.buildpack/some-layer.toml", "[metadata]\n  some-layer-key = \"some-layer-value\""},
+					} {
+						got := h.MustReadFile(t, filepath.Join(layersDir, data.name))
+						h.AssertStringContains(t, string(got), data.want)
+						h.AssertStringDoesNotContain(t, string(got), unsetFlags) // The [types] table shouldn't exist. The build, cache and launch flags are set to false.
+					}
+				})
+
+				when("buildpack API < 0.6", func() {
+					it.Before(func() {
+						analyzer.Buildpacks = []buildpack.GroupBuildpack{
+							{ID: "metadata.buildpack", API: "0.5"},
+							{ID: "no.cache.buildpack", API: "0.5"},
+						}
+					})
+
+					it("restores layer metadata and doesn't change the values of the launch, build and cache flags in top level", func() {
+						h.SkipIf(t, api.MustParse(apiString).Compare(api.MustParse("0.7")) >= 0, "Platform API >= 0.7 does not restore layer metadata")
```

Comment:
Can we move this line to the `it.Before` on line 141?

Reply:
IIRC `SkipIf` has to be inside an `it` block.

Reply:
Interesting. I was able to put it under `it.Before`.